source/actions/ValidationTrackingAction.cc

```diff
@@ -20,12 +20,10 @@
 #include <G4Track.hh>
 #include <G4TrackingManager.hh>
 #include <G4Trajectory.hh>
-#include <G4ParticleDefinition.hh>
 #include <G4OpticalPhoton.hh>
 #include <G4Gamma.hh>
+#include "g4root_defs.hh"
```

Comment:
Same as above

Reply:
These headers pull in the geant4 histogram and ntuples that we are replacing the ROOT histograms with.

Reply:
Histograms and NTuples are created through G4AnalysisManager methods [Analysis Manager Classes](https://geant4-userdoc.web.cern.ch/UsersGuides/ForApplicationDeveloper/html/Analysis/managers.html#files-handling). Therefore whatever in this include is not been used.
The manager chooses the default output file from the filename extension (.csv, .hdf5, .root or .xml). I have tried .csv, .root and .xml and seems to work. The .hdf5 should work if Geant4 is compiled against hdf5.

Reply:
Yep your right, I removed the superfluous header.